arch/sim/src/sim/up_idle.c

```diff
@@ -89,6 +89,15 @@ void up_idle(void)
     }
 #endif
 
+#ifndef CONFIG_SIM_PREEMPTIBLE
+  /* If the system is idle, then process "fake" timer interrupts.
+   * Hopefully, something will wake up.
+   */
+
+  nxsched_process_timer();
```

Comment:
Why CONFIG_SIM_PREEMPTIBLE can't work with CONFIG_SCHED_TICKLESS?

Reply:
We no longer needed the fake timer interrupts delivered from ```Idle Task``` because we can rely on the host timer to drive the scheduler. What do you think ?

Reply:
Yes, but if you support tickless, you should deliver the timer interrupt according OS request. Now you generate the timer interrupt at a fixed interval. Here is more info you can reference:
https://cwiki.apache.org/confluence/display/NUTTX/NuttX+Simulation
That's why I suggest you don't add your own timer implementation, the right thing is enhancing up_oneshot.c which already support both fixed tick and variable tick.

Reply:
> We no longer needed the fake timer interrupts delivered from `Idle Task` because we can rely on the host timer to drive the scheduler. What do you think ?

But we can reuse the most logic inside up_oneshot.c, add the alarm tmer logic and empty up_timer_update.

Reply:
I reused the logic from up_oneshot.c, thanks for the feedback